kale/embed/factorization.py

```diff
@@ -386,31 +394,32 @@ def _fit(self, kernel_x, y, covariates=None):
 
         return self
 
-    def fit_transform(self, x, y=None, covariates=None):
+    def fit_transform(self, x, y=None, groups=None):
         """
         Args:
             x : array-like, shape (n_samples, n_features)
-            y : array-like, shape (n_samples,)
-            covariates : array-like, shape (n_samples, n_covariates)
+            y : array-like, shape (n_labeled_samples,)
+            groups : array-like, shape (n_samples, )
 
         Returns:
             x_transformed : array-like, shape (n_samples, n_components)
         """
-        self.fit(x, y, covariates)
+        self.fit(x, y, groups)
 
-        return self.transform(x, covariates)
+        return self.transform(x, groups)
 
-    def transform(self, x, covariates=None):
+    def transform(self, x, groups=None):
         """
         Args:
             x : array-like, shape (n_samples, n_features)
-            covariates : array-like, augmentation features, shape (n_samples, n_covariates)
+            groups : array-like, augmentation features, shape (n_samples, )
         Returns:
             x_transformed : array-like, shape (n_samples, n_components)
         """
         check_is_fitted(self, "x_fit")
-        if isinstance(covariates, np.ndarray) and self.augmentation:
```

Comment:
before doing anything with the input, I would recommend using the `BaseEstimator`'s `_validate_data` method which has the same arguments as `check_X_y` and `check_array` but the additional thing it has is that it also checks whether or not the input array has the expected shape. An example could be:

```py
def fit(self, X, y=None):
    X = self._validate_data(X, **other_check_array_argument)
    # do stuff
    return self

def transform(self, X):
    X = self._validate_data(X, reset=False)
    # Do stuff
    return X
```

It is necessary to set reset to `False` in `transform` as it means that the input size is already fixed.

Reply:
However if `covariates` or `groups` is already defined, you could use an additional `check_array` and `check_consistent_length`, to ensure the input is valid

Reply:
In this case, I would recommend to also override `_validate_data` to accommodate `groups` or `covariates`